File: backend/src/main/java/pl/dmcs/dto/AddAccountDto.java
package pl.dmcs.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AddAccountDto {

    private String username;
    private String password;
    private String confirmPassword;
    private String email;
    private String firstName;
    private String lastName;
    private boolean active;
    private Set<String> accessLevels;
}

File: frontend/src/dto/AddAccountDto.ts
import { ArrayNotEmpty, IsEmail, IsNotEmpty, MinLength } from 'class-validator';

export class AddAccountDto {

    @IsNotEmpty()
    username: string;

    @MinLength(8)
    password: string;

    @MinLength(8)
    confirmPassword: string;

    @IsEmail()
    email: string;

    @IsNotEmpty()
    firstName: string;

    @IsNotEmpty()
    lastName: string;

    active: boolean;

    @ArrayNotEmpty()
    accessLevels: string[];
}
